### PR TITLE
[development log_in] log_in functionality 

### PR DESCRIPTION
**Pull request by Prakruthee**

### CODE REVIEW

File: api/v1/controllers/account/session.go [hunk 1]
```diff
@@ -18,15 +24,158 @@ var Session sessionController
...
+  }else {
+    db, err := sql.Open("postgres", "password=password host=localhost dbname=online_test_dev sslmode=disable")
+    if err != nil {
+      log.Fatal(err)
+    }
+    get_user_id, err := db.Query("SELECT id, first_name, last_name, email, phone_number, password FROM users WHERE phone_number=$1", l.Phone_number)
+    if err != nil {
+      log.Fatal(err)
+    }
+
```
Comment (praveenmenon) [inline api/v1/controllers/account/session.go]: defer the query

File: api/v1/controllers/account/session.go [hunk 2]
```diff
@@ -18,15 +24,158 @@ var Session sessionController
...
+      err := get_user_id.Scan(&id, &first_name, &last_name, &email, &phone_number, &db_password)
+      if err != nil {
+        log.Fatal(err)
+      }
+
+      check_session, err := db.Query("SELECT user_id from sessions where user_id = $1", id)
+      if err !=nil {
+        log.Fatal(err)
+      }
+
```
Comment (praveenmenon) [inline api/v1/controllers/account/session.go]: same here

File: api/v1/controllers/account/session.go [hunk 3]
```diff
@@ -18,15 +24,158 @@ var Session sessionController
...
+        auth_string := strconv.FormatInt(time.Now().Unix(), 10)
+        h := md5.New()
+        io.WriteString(h, auth_string)
+        auth_token := hex.EncodeToString(h.Sum(nil))
+        var session string = "insert into sessions (start_time, user_id, auth_token) values ($1,$2,$3)"
+        ses, err := db.Prepare(session)
+        if err != nil {
+          log.Fatal(err)
+        }
```
Comment (praveenmenon) [inline api/v1/controllers/account/session.go]: Defer the prepare statement

File: api/v1/controllers/account/session.go [hunk 4]
```diff
@@ -18,15 +24,158 @@ var Session sessionController
...
+        log.Fatal(err)
+      }
+      rw.Header().Set("Content-Type", "application/json")
+      rw.Write(b)
+    }
+  }
+
+  user_login_end: 
 }
+
```
Comment (praveenmenon) [inline api/v1/controllers/account/session.go]: close the db

File: api/v1/controllers/account/session.go (hunk 3 above)
Comment (Prakruthee) [inline api/v1/controllers/account/session.go]: done

**Pull request merged**